use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,484 @@
+```
+
+### Expose and Bind Intel QAT virtual functions to VFIO-PCI (Every reboot)
+
+To enable virtual functions, the host OS should have IOMMU groups enabled. In 
+the UEFI Firmware Intel Virtualization Technology for Directed I/O 
+(Intel VT-d) must be enabled. Also, the kernel boot parameter should be 
+`intel_iommu=on` or `intel_iommu=ifgx_off`. The default in Clear Linux currently 
+is `intel_iommu=igfx_off` which should work with the Intel QAT device. The 
+following commands assume you installed an Intel QAT card, IOMMU is on, and
+VT-d is enabled. The vendor and device ID add to the `VFIO-PCI` driver so that
+each exposed virtual function can be bound to the `VFIO-PCI` driver. Once
+complete, each virtual function passes into a Kata Containers container using
+the PCIe device passthrough feature. For Kubernetes, the Intel device plugin
+for Kubernetes handles the binding of the driver but the VF’s still must be
+enabled.
+
+```sh
+$ sudo modprobe vfio-pci
+$ QAT_PCI_BUS_PF_NUMBERS=$(lspci | grep Quick | cut -d ' ' -f 1)
+$ QAT_PCI_BUS_PF_1=$(echo $QAT_PCI_BUS_PF_NUMBERS | cut -d ' ' -f 1)
+$ echo 16 | sudo tee /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/sriov_numvfs
+$ QAT_PCI_ID_VF=$(cat /sys/bus/pci/devices/0000:${QAT_PCI_BUS_PF_1}/virtfn0/uevent | grep PCI_ID)
+$ QAT_VENDOR_AND_ID_VF=$(echo ${QAT_PCI_ID_VF/PCI_ID=} | sed 's/:/ /')
+$ echo $QAT_VENDOR_AND_ID_VF | sudo tee --append /sys/bus/pci/drivers/vfio-pci/new_id
+```
+Loop through all the virtual functions and bind to the VFIO driver
+```sh
+$ for f in /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/virtfn*
+  do QAT_PCI_BUS_VF=$(basename $(readlink $f))
+   echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/c6xxvf/unbind
+   echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/vfio-pci/bind
+  done
+```
+
+### Check Intel QAT virtual functions are enabled
+
+If the following command returns empty, then the virtual functions are not 
+properly enabled. This command checks the enumerated device IDs for just the 
+virtual functions. Using the Intel QAT as an example, the physical device ID 
+is `37c8` and virtual function device ID is `37c9`. The following command checks 
+if VF's are enabled for any of the currently known Intel QAT device ID's. The
+following `ls` command should show the 16 VF's bound to `VFIO-PCI`.
+
+```sh
+$ for i in 0442 0443 37c9 19e3; do lspci -d 8086:$i; done
+```
+
+Another way to check is to see what PCI devices that `VFIO-PCI` is mapped to.
+It should match the device ID's of the VF's.
+```sh
+$ ls -la /sys/bus/pci/drivers/vfio-pci
+```
+
+## Prepare Kata Containers
+
+### Download Kata Kernel Source
+
+This example automatically uses the latest kernel supported by Kata. It 
+follows the instructions from the
+[packaging kernel repository](https://github.com/kata-containers/packaging/tree/master/kernel)
+and uses the latest kata-kernel 
```

Comment:
I think this should probably be, "the latest Kata Containers kernel" (or "the latest Kata kernel"). It looks like this might be trying to be a package name (?), but the kernel is in fact contained in the `kata-linux-container` package.

Reply:
I removed the hyphen. I also noticed I was not consistent in capitalizing/not capitalizing kernel throughout the doc so I made it all lowercase throughout.